src/gui/plugins/plotting/Plotting.cc

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <ignition/plugin/Register.hh>
+
+#include "Plotting.hh"
+
+namespace ignition::gazebo
+{
+  class PlottingPrivate
+  {
+    /// \brief Interface to communicate with Qml
+    public: ignition::gui::PlottingInterface *plottingIface;
+
+    public: std::map<std::string, PlotComponent*> components;
+  };
+
+  class PlotComponentPrivate
+  {
+    /// \brief entity id in the simulation
+    public: uint64_t entity;
+
+    /// \brief type identifier unique to each component type
+    public: uint64_t typeId;
+
+    /// \brief component data type (Pose3d, Vector3d, double)
+    public: std::string type;
+
+    /// \brief attributes of the components,
+    /// ex: x,y,z attributes in Vector3d type component
+    public: std::map<std::string, ignition::gui::PlotData*> data;
+  };
+}
+
+using namespace ignition::gazebo;
+using namespace ignition::gui;
+
+//////////////////////////////////////////////////
+PlotComponent::PlotComponent(std::string _type, uint64_t _entity,
+                             uint64_t _typeId) :
+    dataPtr(new PlotComponentPrivate)
+{
+  this->dataPtr->entity = _entity;
+  this->dataPtr->typeId = _typeId;
+  this->dataPtr->type = _type;
+
+  if (_type == "Vector3d")
+  {
+    this->dataPtr->data["x"] = new PlotData();
+    this->dataPtr->data["y"] = new PlotData();
+    this->dataPtr->data["z"] = new PlotData();
+  }
+  else if (_type == "Pose3d")
+  {
+    this->dataPtr->data["x"] = new PlotData();
+    this->dataPtr->data["y"] = new PlotData();
+    this->dataPtr->data["z"] = new PlotData();
+    this->dataPtr->data["roll"] = new PlotData();
+    this->dataPtr->data["pitch"] = new PlotData();
+    this->dataPtr->data["yaw"] = new PlotData();
+  }
+  else if (_type == "double")
+      this->dataPtr->data["value"] = new PlotData();
+  else
+    ignwarn << "Invalid Plot Component Type:" << _type << std::endl;
+}
+
+//////////////////////////////////////////////////
+void PlotComponent::RegisterChart(std::string _attribute, int _chart)
+{
+  if (this->dataPtr->data.count(_attribute) == 0)
+  {
+    ignwarn << "Invalid Plot Component Attribute" << std::endl;
+    return;
+  }
+  this->dataPtr->data[_attribute]->AddChart(_chart);
```

Comment:
You should check if `this->dataPtr->data.find(_attribute) != this->dataPtr->data.end()` before calling `AddChart`.

This also applies to other places.

Reply:
yeah i already do that, in that if condition above it (and all other places)